core/app/models/spree/taxonomy.rb

```diff
@@ -25,6 +25,8 @@ def set_root
     end
 
     def set_root_taxon_name
+      return unless saved_change_to_name?
```

Comment:
@MatthewKennedy let's add some test for this please :)

Reply:
@damianlegawiec Will do.